File: ruby/cosasDesorganizadas/matriz.rb
# encoding: utf-8


def getMatriz
	matriz = [
			[1,2,3],
			[4,5,6],
			[7,8,9]]

	return matriz
end

def imprimirUpper(matriz,j)
	
	for i in 0..j do
		print matriz[i][j].to_s+' '
		j = j-1
	end
	p ''
end

def imprimirLower(matriz,j)
	i = matriz.length-1
	for j in j..matriz.length-1 do
		print matriz[j][i].to_s+' '
		i = i-1
	end
	p ''
end

def imprimirDiagonal(matriz)
	matriz.length.times{ |i| imprimirUpper(matriz,i)}
	
	for i in 1..matriz.length do
		imprimirLower(matriz,i)
	end

	
end

if __FILE__ == $0
	matriz = getMatriz

	imprimirDiagonal(matriz)
end

File: python/cosasDesorganizadas/matriz.py
import sys

def getMatriz():
	matriz = [
			[1,2,3,4],
			[2,9,5,2],
			[3,7,3,6],
			[1,2,1,2]
		  ]
	return matriz

def imprimirUpper(matriz,n):
	j = 0
	
	for i in reversed(range(0,n+1)):
		sys.stdout.write( str( matriz[j][i] )+' ' )
		j = j+1
	print

def imprimirLower(matriz,n):
	j = len(matriz)-1
	
	for i in range(n,len(matriz) ):
		
		sys.stdout.write( str( matriz[i][j] )+' ' )
		j = j-1
	print	

def imprimirDiagonal(matriz):
	for i in range(len(matriz)):
		imprimirUpper(matriz,i)

	for i in range(1,len(matriz)):
		imprimirLower(matriz,i)


if __name__ == "__main__":
	matriz = getMatriz()
	imprimirDiagonal(matriz)
